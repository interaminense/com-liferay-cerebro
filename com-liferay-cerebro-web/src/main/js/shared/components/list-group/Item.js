import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

class Item extends Component {
	render() {
		const {children, flex, header} = this.props;

		const classes = getCN({
			'list-group-header': header,
			'list-group-item': !header,
			'list-group-item-flex': flex
		});

		return (
			<li {...this.otherProps()} class={classes}>
				{children}
			</li>
		);
	}
}

Item.PROPS = {
	flex: Config.bool().value(false),
	header: Config.bool().value(false)
};

export default Item;