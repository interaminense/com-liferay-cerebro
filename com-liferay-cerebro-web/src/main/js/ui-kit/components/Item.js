import Component from 'metal-jsx';

class Item extends Component {
	render() {
		return <div class="kit-item-root">{this.props.children}</div>;
	}
}

export default Item;