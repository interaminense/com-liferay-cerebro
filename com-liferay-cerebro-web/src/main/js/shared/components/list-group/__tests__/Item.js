import Item from '../Item';

describe('Item', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Item();

		expect(component).toMatchSnapshot();
	});
});