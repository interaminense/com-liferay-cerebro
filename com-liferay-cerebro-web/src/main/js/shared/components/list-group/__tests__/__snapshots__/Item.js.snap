// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render 1`] = `<li class="list-group-item"></li>`;
